### DISCUSSION TITLE
SimpleTriggerImpl: should TimesTriggered actually be virtual?

### DISCUSSION THREAD

**Discussion by drieseng** [quartznet/quartznet#1398] [General]
We sometimes access the underlying backing field for this property, so overriding this property will not have the desired effect. 

I propose to either update our implementation to always use the property instead of directly accessing the backing field (in which case there will be more virtual calls), or - which is what I prefer - no longer mark this property virtual in **v4**.

@lahma, what do you think?

**Comment by lahma** [comment]
Can be non-virtual. I might even go one step further and seal the triggers in v4 and open them again if someone really has a valid use case to inherit directly from them.
